pkg/compactor/blocks_cleaner.go

```diff
@@ -638,3 +667,73 @@ func stalePartialBlockLastModifiedTime(ctx context.Context, blockID ulid.ULID, u
 	}
 	return lastModified, err
 }
+
+func (c *BlocksCleaner) computeCompactionJobs(ctx context.Context, userID string, userBucket objstore.InstrumentedBucket, idx *bucketindex.Index) (int, int, error) {
+	metas := convertBucketIndexToMetasForCompactionJobPlanning(idx)
+
+	// We need to pass this metric to Filters, but we don't need to report this value from BlocksCleaner.
+	synced := newNoopGaugeVec()
+
+	for _, f := range []block.MetadataFilter{
+		// We don't include ShardAwareDeduplicateFilter, because thus filter relies on list of compaction sources, which are not present in the BucketIndex.
+		// We do include NoCompactionMarkFilter to avoid computing jobs from blocks that are marked for no-compaction.
+		NewNoCompactionMarkFilter(userBucket, true),
```

Comment:
Unrelated: in a follow up PR I would be glad to see the `NewNoCompactionMarkFilter ()` bool param to be removed, because we always pass `true` in Mimir (the flag was introduced in Thanos, because Thanos didn't remove metas of blocks marked for no compaction).

Reply:
https://github.com/grafana/mimir/pull/7301